File: Natixis Craftsmanship Practices/Breaking Dependencies/MarketTrading/csharp/External/NativeMarketSql.cs
namespace MarketTrading.External;

[DotNotModify]
public class NativeMarketSql
{
    public string LoadPositionAsXml(string trader, string symbol) {
        throw new ExternalSideEffectException("SQL XML load forbidden in test");
    }

    public void PersistTradeAsXml(string tradeXml) {
        throw new ExternalSideEffectException("SQL XML persist forbidden in test");
    }
}

File: Natixis Craftsmanship Practices/Breaking Dependencies/MarketTrading/cpp/external/src/NativeMarketSql.cpp
#include "NativeMarketSql.h"
#include "ExternalSideEffectException.h"

std::string NativeMarketSql::loadPositionAsXml(const std::string &, const std::string &) {
    throw ExternalSideEffectException("SQL XML load forbidden in test");
}

void NativeMarketSql::persistTradeAsXml(const std::string &) {
    throw ExternalSideEffectException("SQL XML persist forbidden in test");
}
